src/cmd/flash.js

```diff
@@ -55,11 +59,108 @@ module.exports = class FlashCommand extends CLICommandBase {
 		} else if (local) {
 			let allFiles = binary ? [binary, ...files] : files;
 			await this.flashLocal({ files: allFiles, applicationOnly, target });
+		} else if (tachyon) {
+			let allFiles = binary ? [binary, ...files] : files;
+			await this.flashTachyon({ verbose, files: allFiles });
 		} else {
 			await this.flashCloud({ device, files, target });
 		}
 	}
 
+	async flashTachyon({ verbose, files }) {
+		this.ui.write(`${os.EOL}Ensure only one device is connected to the computer${os.EOL}`);
+
+		let unpackToolFolder;
+		if (files.length === 0) {
+			// If no files are passed, assume the current directory
+			unpackToolFolder = process.cwd();
+			files = await fs.readdir(unpackToolFolder);
+		} else if (files.length === 1) {
+			// If only one file is passed, check if it's a directory
+			const stats = await fs.stat(files[0]);
+			if (stats.isDirectory()) {
+				unpackToolFolder = files[0];
+				files = await fs.readdir(files[0]);
+			}
+		} else {
+			// If multiple files are passed, check the directory from the first file
+			unpackToolFolder = path.dirname(files[0]);
+		}
+
+		let linxuFiles = await this._findLinuxExecutableFiles(files, { directory: unpackToolFolder });
+		linxuFiles = linxuFiles.map(f => path.basename(f));
+
+		const elfFiles = linxuFiles.filter(f => f.startsWith('prog') && f.endsWith('.elf'));
```

Comment:
Change the criteria to
```suggestion
		const elfFiles = linxuFiles.filter(f => f.includes('firehose') && f.endsWith('.elf'));
```
This matches https://www.notion.so/particle/PRD-Particle-CLI-Support-for-Flashing-Tachyon-157d40caa5dc805ab72fdabddf713381?pvs=4#158d40caa5dc80db8131ecd541a48d17